src/main/java/net/unicon/lti/service/lti/impl/RegistrationServiceImpl.java

```diff
@@ -142,20 +142,24 @@ public ToolRegistrationDTO generateToolConfiguration(PlatformRegistrationDTO pla
         //OPTIONAL -->setSecondary_domains --> Collections.singletonList
         //OPTIONAL -->setDeployment_id
 
-        toolConfigurationDTO.setTarget_link_uri(domainUrl);
+        toolConfigurationDTO.setTarget_link_uri(localUrl + TextConstants.LTI3_SUFFIX);
 
         //OPTIONAL -->setCustom_parameters --> Map
         toolConfigurationDTO.setDescription(description);
         List<ToolMessagesSupportedDTO> messages = new ArrayList<>();
 
-        // Indicate Deep Linking support
-        ToolMessagesSupportedDTO message1 = new ToolMessagesSupportedDTO();
-        message1.setType("LtiDeepLinkingRequest");
-        message1.setTarget_link_uri(localUrl + TextConstants.LTI3_SUFFIX);
-        message1.setLabel(deepLinkingMenuLabel);
-//        OPTIONAL: --> message1 --> setIcon_uri
-//        OPTIONAL: --> message1 --> setCustom_parameters
-        messages.add(message1);
+        // Staging: https://goldilocks.ludev.team 
+        // Production: https://goldilocks.lumenlearning.com 
+        if (!domainUrl.contains("goldilocks")) { // Goldilocks shouldn't support deep linking - although this is non-critical and will not break dynamic registration if deep linking IS set for Goldilocks
```

Comment:
This will be okay for now but long term, it's not ideal.  I almost wonder if we need an environment variable to say that deep linking is turned on for an instance of the middleware.

Reply:
We have the `lti13.enableDeepLinking` environment variable, so the value from that could be pulled in on this line instead.

Reply:
That's a good idea - thank you Mary!  Matthew, could we do that?